src/4D_api.cpp

```diff
@@ -189,6 +189,11 @@ double proj_roundtrip(PJ *P, PJ_DIRECTION direction, int n, PJ_COORD *coord) {
     /* finally, we take the last half-step */
     t = proj_trans(P, opposite_direction(direction), t);
 
+    /* if we start with NaN, we expect NaN as output */
+    if (isnan(org.v[0]) && isnan(t.v[0])) {
```

Comment:
Should I check all 4 coordinates?

Reply:
> Should I check all 4 coordinates?

I think that would make sense. Also, should it be changed to `or` instead of `and`?

Reply:
Regarding the or/and: I don't think so. This is checking that the original input is NaN *and* the final output is NaN. If the final output was NaN, but we didn't give it something that should return NaN then that should be a failure.

Reply:
I could do: `(isnan(org.v[0]) || isnan(org.v[1]) || ...) && (isnan(t.v[0]) || isnan(t.v[1]) || ...)`? Basically if any of the inputs are NaN and any of the outputs are NaN...no maybe it should be that any NaNs in the input produce all NaNs in the output?

Reply:
If everyone agrees on other parts of this PR, I think this is the only other part I might want to change. That is, checking all axes for NaN. This also adds to execution time.

Reply:
proj_roundtrip() makes only sense as part of a test utility like gie, not "production" use cases. Execution time is not a concern